cypress/App.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react';
+import 'bulma/css/bulma.css';
+import './App.scss';
+
+export const goodsFromServer = [
+  'Dumplings',
+  'Carrot',
+  'Eggs',
+  'Ice cream',
+  'Apple',
+  'Bread',
+  'Fish',
+  'Honey',
+  'Jam',
+  'Garlic',
+  'Tomatoes',
+];
+
+const prepareGoods = (goods, { field, isReverse }) => {
+  const sortedGoods = [...goods];
+
+  if (field === 'alphabetical') {
+    sortedGoods.sort();
+  } else if (field === 'length') {
+    sortedGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReverse) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortField, setSortField] = useState(null);
+  const [originalGoods] = useState(goodsFromServer);
+
+  const handleSortByName = () => {
+    setSortField('alphabetical');
+    setVisibleGoods(
+      prepareGoods(goodsFromServer, {
+        field: 'alphabetical',
+        isReverse: isReversed,
+      }),
```

Comment:
The `prepareGoods` function is being called with `goodsFromServer` instead of `visibleGoods`. This means the current state of `visibleGoods` is not considered when sorting, which might not be the intended behavior.